### PR TITLE
PLT-2260 Add the ability to switch from email to ldap and back

### PR DESCRIPTION
**Pull request by jwilander**
Refactored the claim stuff so it was implemented in a react-router friendly manner.

There is an enterprise PR to go along this.

### CODE REVIEW

File: api/user.go [hunk 1]
```diff
@@ -2230,6 +2233,176 @@ func switchToEmail(c *Context, w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	RevokeAllSession(c, user.Id)
+	if c.Err != nil {
+		return
+	}
+
+	ldapInterface := einterfaces.GetLdapInterface()
+	if ldapInterface == nil {
```
Comment (coreyhulen) [inline api/user.go]: I'm starting to see this pattern everywhere.  I wonder if we should create a dev driven item to refactor?  Also shouldn't you check to make sure they have the LDAP feature enabled and their license is valid?

Comment (jwilander) [inline api/user.go]: Inside the enterprise repo the chunk of code that connects to the LDAP server checks the license and the config setting. Agree that it's probably a good idea to refactor it so we don't need to paste these few lines in every time, maybe implement it as a part of the interface

**Comment by coreyhulen** [comment]
+1

File: api/user.go [hunk 2]
```diff
@@ -67,6 +65,11 @@ func InitUser(r *mux.Router) {
 	sr.Handle("/{id:[A-Za-z0-9]+}/sessions", ApiUserRequired(getSessions)).Methods("GET")
 	sr.Handle("/{id:[A-Za-z0-9]+}/audits", ApiUserRequired(getAudits)).Methods("GET")
 	sr.Handle("/{id:[A-Za-z0-9]+}/image", ApiUserRequired(getProfileImage)).Methods("GET")
+
```
Comment (enahum) [inline api/user.go]: I think it would be better to have this routes inside a api/claim.go file and under /api/v1/claim route.

Comment (jwilander) [inline api/user.go]: Hmmm, but these APIs are actions on the user object themselves. The user API file is getting a bit crowded though, so this is probably worth a discussion. Propose no action here, but discuss during the next dev meeting?

File: webapp/components/claim/components/email_to_oauth.jsx [hunk 3]
```diff
@@ -1,26 +1,14 @@
...
-    pwdError: {
-        id: 'claim.email_to_sso.pwdError',
-        defaultMessage: 'Please enter your password.'
-    },
-    pwd: {
-        id: 'claim.email_to_sso.pwd',
-        defaultMessage: 'Password'
-    }
-});
+import * as Utils from '../../../utils/utils.jsx';
```
Comment (crspeller) [inline webapp/components/claim/components/email_to_oauth.jsx]: Unnecessary long paths. You only need `utils/utils.jsx`. Webpack considers everything at top level a module.

**Comment by crspeller** [comment]
+1

**Pull request merged**